lib/logstash/inputs/tcp.rb

```diff
@@ -115,7 +115,6 @@ def run_server(output_queue)
       rescue OpenSSL::SSL::SSLError => e
         # log error, close socket, accept next connection
         @logger.error("SSL Error", :exception => e, :backtrace => e.backtrace)
-        socket.close rescue nil
```

Comment:
I comfort myself poorly with this kind of error because any language with "unassigned value"-kind of detection at compile-time would have caught this error :|